### PR TITLE
Add three letter abbreviations to ProteinAlphabet

### PR DESCRIPTION
**Pull request by acesnik**

### CODE REVIEW

**Review by evolvedmicrobe — CHANGES_REQUESTED** [review]
Hey Anthony,

Welcome to .NET Bio and thanks for submitting the pull request!  I looked it over, and am not entirely sure I get what the intention was here so left some comments/questions for you.  Adding three letter codes certainly seems useful though so I am glad to see that in here.  Anyway, let me know your responses to comments and we'll go from there!

Cheers,
Nigel

File: Source/Bio.Core/AmbiguousProteinAlphabet.cs [hunk 1]
```diff
@@ -38,10 +38,10 @@ protected AmbiguousProteinAlphabet()
             this.B = (byte)'B';
             this.J = (byte)'J';
 
-            AddAminoAcid(this.X, "Undetermined or atypical", (byte)'x');
-            AddAminoAcid(this.Z, "Glutamic or Glutamine", (byte)'z');
-            AddAminoAcid(this.B, "Aspartic or Asparagine", (byte)'b');
-            AddAminoAcid(this.J, "Leucine or Isoleucine", (byte)'j');
+            AddAminoAcid(this.X, "Unx", "Undetermined or atypical", (byte)'x');
```
[the remark this answers is not in the document]
Comment (acesnik) [reply Source/Bio.Core/AmbiguousProteinAlphabet.cs]: Ah! Thanks for catching that; those were informal abbreviations. I updated them to Xaa, etc. in the latest push.

File: Source/Bio.Core/ProteinAlphabet.cs [hunk 2]
```diff
@@ -63,6 +63,11 @@ public class ProteinAlphabet : IAlphabet
         /// </summary>
         private readonly Dictionary<byte, byte> aminoAcidValueMap = new Dictionary<byte, byte>();
 
+        /// <summary>
+        /// Symbol to three amino acid abbreviation.
+        /// </summary>
+        private readonly Dictionary<byte, string> threeLetterAbbreviationMap = new Dictionary<byte, string>();
```
[the remark this answers is not in the document]
Comment (acesnik) [reply Source/Bio.Core/ProteinAlphabet.cs]: Good point on the variable being private. My intention was to allow the user to get 3-letter abbreviations from a symbol, and vice versa. I added get methods to satisfy that intention.

**Comment by acesnik** [comment]
Thanks for the welcome! I responded to your comments and made some changes accordingly.

**Comment by evolvedmicrobe** [comment]
Looks good to me, congrats on first commit!

Going forward, you may find it easier to name branches based on the types of feature they have (e.g. `git checkout -b threeletter` to make a branch for this feature), and it can also be nice to squash your commits into one later on.  For a first go though, no need to worry about this.

All the best,
Nigel

**Review by evolvedmicrobe — APPROVED** [review]

**Comment by acesnik** [comment]
Thanks! 

Sounds good. I'll make feature branches and squash commits in the future.

Anthony

**Pull request merged**